models/image.py

```diff
@@ -1720,6 +1758,65 @@ def score(self):
     def score(self, value):
         self._score = value
 
+    @property
+    def psfflux(self):
+        """
+        The flux of the PSF (2D float array).
+        """
+        if self._data is None and self.filepath is not None:
+            self.load()
+        return self._psfflux
+
+    @psfflux.setter
+    def psfflux(self, value):
+        self._psfflux = value
+
+    @property
+    def psffluxerr(self):
+        """
+        The error on the flux of the PSF (2D float array).
+        """
+        if self._data is None and self.filepath is not None:
+            self.load()
+        return self._psffluxerr
+
+    @psffluxerr.setter
+    def psffluxerr(self, value):
+        self._psffluxerr = value
+
+    @property
+    def nandata(self):
+        """The image data, only masked with NaNs wherever the flag is not zero. """
+        if self._nandata is None:
+            self._nandata = self.data.copy()
+            if self.flags is not None:
+                self._nandata[self.flags != 0] = np.nan
+        return self._nandata
+
+    @nandata.setter
+    def nandata(self, value):
+        self._nandata = value
+
+    def show(self, **kwargs):
+        """
+        Display the image using the matplotlib imshow function.
+
+        Parameters
+        ----------
+        **kwargs: passed on to matplotlib.pyplot.imshow()
+            Additional keyword arguments to pass to imshow.
+        """
+        import matplotlib.pyplot as plt
+        mu, sigma = sigma_clipping(self.data)
```

Comment:
I'll have to look at what you did with sigma-clipping.  I've used astropy.visualization.ZScaleInterval for this kind of thing in the past.

On other fronts: part of me thinks that this doesn't belong in the image class.  It's a view thing, not a model thing.

Reply:
I am thinking object-oriented rather than database mapping... having each object able to draw itself on the figure is good OOP but I'm not against having a more serious plot function somewhere else. Then it would still need to be object-specific (draw an Image is not the same as draw Cutouts).